atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestTable.java

```diff
@@ -688,21 +691,31 @@ public void delete(Iterable<SchemaApiTestRow> rows) {
         return transformed;
     }
 
-    public BatchingVisitableView<SchemaApiTestRowResult> getRange(RangeRequest range) {
+    private RangeRequest augmentRangeRequest(RangeRequest range) {
         if (range.getColumnNames().isEmpty()) {
-            range = range.getBuilder().retainColumns(allColumns).build();
+            return range.getBuilder().retainColumns(allColumns).build();
         }
-        return BatchingVisitables.transform(t.getRange(tableRef, range), new Function<RowResult<byte[]>, SchemaApiTestRowResult>() {
+        return range;
+    }
+
+    public BatchingVisitableView<SchemaApiTestRowResult> getRange(RangeRequest range) {
+        return BatchingVisitables.transform(t.getRange(tableRef, augmentRangeRequest(range)), new Function<RowResult<byte[]>, SchemaApiTestRowResult>() {
             @Override
             public SchemaApiTestRowResult apply(RowResult<byte[]> input) {
                 return SchemaApiTestRowResult.of(input);
             }
         });
     }
 
+    private Iterable<RangeRequest> augmentRanges(Iterable<RangeRequest> ranges) {
+        return StreamSupport.stream(ranges.spliterator(), false)
```

Comment:
generally, prefer having the method name tell you what it does. Augment just means 'change', pretty much - maybe something like `optimizeColumnSelections` and `optimizeColumnSelection`.

Also, instead of streaming it and collecting back into a new collection, I might be minded to just write `return Iterables.transform(ranges, this::optimizeColumnSelection)`.